Objects/unicodeobject.c

```diff
@@ -7422,8 +7422,11 @@ decode_code_page_stateful(int code_page,
     do
     {
 #ifdef NEED_RETRY
-        if (size > INT_MAX) {
-            chunk_size = INT_MAX;
+        /* INT_MAX is the theoretical largest chunk, but INT_MAX / 4
+           performs better and avoids partial characters overrunning the
+           length limit in MultiByteToWideChar on Windows */
+        if (size > INT_MAX / 4) {
+            chunk_size = INT_MAX / 4;
```

Comment:
It seems that `INT_MAX / 4` is done in quite a few places. IMHO, it would be nice to use a constant.